src/Http/Middleware/ErrorHandlerMiddleware.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Middleware;
+
+use Cake\Core\App;
+use Cake\Http\ResponseTransformer;
+use Exception;
+
+/**
+ * Error handling middleware.
+ *
+ * Traps exceptions and converts them into HTML or content-type appropriate
+ * error pages using the CakePHP ExceptionRenderer.
+ */
+class ErrorHandlerMiddleware
+{
+    /**
+     * Constructor
+     *
+     * @param string|callable $renderer The renderer or class name
+     *   to use or a callable factory.
+     */
+    public function __construct($renderer = null)
+    {
+        $this->renderer = $renderer ?: 'Cake\Error\ExceptionRenderer';
+    }
+
+    /**
+     * Wrap the remaining middleware with error handling.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request.
+     * @param \Psr\Http\Message\ResponseInterface $response The response.
+     * @param callable $next Callback to invoke the next middleware.
+     * @return \Psr\Http\Message\ResponseInterface A response
+     */
+    public function __invoke($request, $response, $next)
+    {
+        try {
+            return $next($request, $response);
+        } catch (\Exception $e) {
+            return $this->handleException($e, $request, $response);
+        }
+    }
+
+    /**
+     * Handle an exception and generate an error response
+     *
+     * @param \Exception $exception The exception to handle.
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request.
+     * @param \Psr\Http\Message\ResponseInterface $response The response.
+     * @return \Psr\Http\Message\ResponseInterface A response
+     */
+    public function handleException($exception, $request, $response)
+    {
+        $renderer = $this->getRenderer($exception);
+        try {
+            $response = $renderer->render();
+            return ResponseTransformer::toPsr($response);
+        } catch (Exception $e) {
+            $message = sprintf(
+                "[%s] %s\n%s", // Keeping same message format
+                get_class($e),
+                $e->getMessage(),
+                $e->getTraceAsString()
+            );
+            trigger_error($message, E_USER_ERROR);
+        }
+        return $response;
```

Comment:
We could augment the response status code to a 50x and set a text response in addition to triggering an error.